controllers/spannerautoscaler_controller.go

```diff
@@ -341,49 +301,44 @@ func (r *SpannerAutoscalerReconciler) needUpdateProcessingUnits(sa *spannerv1bet
 	}
 }
 
-// For testing purpose only
-func calcDesiredNodes(currentCPU, currentNodes, targetCPU, minNodes, maxNodes, scaledownStepSize int) int {
-	return calcDesiredProcessingUnits(currentCPU, currentNodes*1000, targetCPU, minNodes*1000, maxNodes*1000, scaledownStepSize) / 1000
-}
-
-// nextValidProcessingUnits finds next valid value in processing units.
-// https://cloud.google.com/spanner/docs/compute-capacity?hl=en
-// Valid values are
-// If processingUnits < 1000, processing units must be multiples of 100.
-// If processingUnits >= 1000, processing units must be multiples of 1000.
-func nextValidProcessingUnits(processingUnits int) int {
-	if processingUnits < 1000 {
-		return ((processingUnits / 100) + 1) * 100
-	}
-	return ((processingUnits / 1000) + 1) * 1000
-}
-
-func maxInt(first int, rest ...int) int {
-	result := first
-	for _, v := range rest {
-		if result < v {
-			result = v
-		}
-	}
-	return result
-}
-
 // calcDesiredProcessingUnits calculates the values needed to keep CPU utilization below TargetCPU.
-func calcDesiredProcessingUnits(currentCPU, currentProcessingUnits, targetCPU, minProcessingUnits, maxProcessingUnits, scaledownStepSize int) int {
-	totalCPUProduct1000 := currentCPU * currentProcessingUnits
+func calcDesiredProcessingUnits(sa spannerv1beta1.SpannerAutoscaler) int {
+	totalCPU := sa.Status.CurrentHighPriorityCPUUtilization * sa.Status.CurrentProcessingUnits
+
+	requiredPU := totalCPU / sa.Spec.ScaleConfig.TargetCPUUtilization.HighPriority
+
+	var nextPU int
+
+	// https://cloud.google.com/spanner/docs/compute-capacity?hl=en
+	// Valid values for processing units are:
+	// If processingUnits < 1000, processing units must be multiples of 100.
+	// If processingUnits >= 1000, processing units must be multiples of 1000.
+	//
+	// Round up the requiredPU value to make it valid
+	// If it is already a valid PU, increment to next unit to keep CPU usage below desired threshold
+	if requiredPU < 1000 {
+		nextPU = ((requiredPU / 100) + 1) * 100
+	} else {
+		nextPU = ((requiredPU / 1000) + 1) * 1000
+	}
 
-	desiredProcessingUnits := maxInt(nextValidProcessingUnits(totalCPUProduct1000/targetCPU), currentProcessingUnits-scaledownStepSize*1000)
+	desiredPU := nextPU
```

Comment:
It looks like to just use the `nextPU` without the `desiredPU`.

Reply:
Yes, thanks! I changed this now.